core/src/test/resources/org/apache/calcite/test/SqlToRelConverterTest.xml

```diff
@@ -6513,12 +6552,11 @@ LIMIT 5]]>
 LogicalSort(sort0=[$2], dir0=[ASC], fetch=[5])
   LogicalAggregate(group=[{0, 1, 2, 3}])
     LogicalProject(EMPNO=[$0], ENAME=[$1], DEPTNO=[$2], RANK_VAL=[$3])
-      LogicalFilter(condition=[$5])
-        LogicalProject(EMPNO=[$0], ENAME=[$1], DEPTNO=[$2], RANK_VAL=[$3], EXPR$0=[$4], QualifyExpression=[=($3, $4)])
+      LogicalFilter(condition=[$4])
+        LogicalProject(EMPNO=[$0], ENAME=[$1], DEPTNO=[$7], RANK_VAL=[RANK() OVER (PARTITION BY $1 ORDER BY $7 DESC)], QualifyExpression=[=(RANK() OVER (PARTITION BY $1 ORDER BY $7 DESC), $9)])
```

Comment:
I think this is a performance regression, since we are computing the window function multiple times? I believe this is the scenario that the comment "This is a very specific application of Common Subexpression Elimination" was trying to handle.

Reply:
I made note of this regression, and I'd rather it had not happened, but I don't view it as very serious. `Project` (unlike `Calc`) is not able to represent common subexpressions, so the appearance of duplicate expressions in `Project` does not mean that they are going to be executed separately.

A reasonable planner configuration will do subexpression elimination. Sql-to-rel shouldn't try too hard to do that.

Reply:
Good point!